### PR TITLE
NXP-20274: fix polling of asynchronous conversion

### PR DESCRIPTION
**Pull request by troger**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/nuxeo/nuxeo/pull/192" target="_blank">ReviewNinja

File: nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/adapters/ConvertAdapter.java [hunk 1]
```diff
@@ -180,12 +180,10 @@ public Object convert(@FormParam("converter") String converter, @FormParam("type
             throw new IllegalParameterException("No converter, type or format parameter specified");
         }
 
-        String serverURL = ctx.getServerURL().toString();
-        if (serverURL.endsWith("/")) {
-            serverURL = serverURL.substring(0, serverURL.length() - 1);
-        }
+        String serverURL = ctx.getServerURL().toString().replaceAll("/$", "");
```
Comment (efge) [inline nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/adapters/ConvertAdapter.java]: I'd use `StringUtils.stripEnd` instead maybe.

**Comment by efge** [comment]
👍

File: nuxeo-features/rest-api/nuxeo-rest-api-io/src/main/java/org/nuxeo/ecm/restapi/jaxrs/io/conversion/ConversionScheduled.java [hunk 2]
```diff
@@ -28,8 +28,11 @@
 
     public final String pollingURL;
 
-    public ConversionScheduled(String id, String pollingURL) {
+    public final String resultURL;
```
Comment (atchertchian) [inline nuxeo-features/rest-api/nuxeo-rest-api-io/src/main/java/org/nuxeo/ecm/restapi/jaxrs/io/conversion/ConversionScheduled.java]: maybe keep the old constructor, deprecated, for backport?

File: nuxeo-core/nuxeo-core-cache/src/main/java/org/nuxeo/ecm/core/transientstore/work/TransientStoreWork.java [hunk 3]
```diff
@@ -46,6 +46,10 @@
 
     protected String entryKey;
 
+    public static String computeEntryKey(String id) {
+        return id + KEY_SUFFIX;
```
Comment (atchertchian) [inline nuxeo-core/nuxeo-core-cache/src/main/java/org/nuxeo/ecm/core/transientstore/work/TransientStoreWork.java]: missing since version?

**Comment by atchertchian** [comment]
:+1:

**Comment by troger** [comment]
Merged 2dd2f93632de343d55f59c55c6a9767b4fa10303

**Pull request closed without merging**